### PR TITLE
Expose `lam` and `momentum` in the appropriate kilosorts

### LINKED ISSUE #2352: Missing "lam" and "momentum" Kilosort parameters
Hi,

Spotted that "lam" and "momentum" parameters for Kilosort are hardcoded rather than accessible from default_params dictionary. Would it be possible to add them?
e.g. in spikeinterface/src/spikeinterface/sorters/external/kilosort2.py

### PR DESCRIPTION
**Pull request by zm711**
Fixes #2352 

Changes:

1) lam and momentum in 2 & 2.5
2) lam in 3

I didn't touch kilosort1 because the lam and momentum are given slightly differently and they are hardcoded with defaults that are different from the kilosort1 defaults. Is there a reason for this? Should I expose those or just wait for a request?

### CODE REVIEW

File: src/spikeinterface/sorters/external/kilosort2.py [hunk 1]
```diff
@@ -162,16 +166,17 @@ def _get_specific_options(cls, ops, params):
...
-        ops["lam"] = 10.0
+        ops["lam"] = params["lam"]
 
         # splitting a cluster at the end requires at least this much isolation for each sub-cluster (max = 1)
         ops["AUCsplit"] = params["AUCsplit"]
 
         # minimum spike rate (Hz), if a cluster falls below this for too long it gets removed
         ops["minFR"] = params["minFR"]
 
+        momentum = [float(mom) for mom in params["momentum"]]
```
Comment (zm711) [inline src/spikeinterface/sorters/external/kilosort2.py:177]: The same was done above for `threshold_projection`, but I don't know exactly why so I did this just to be safe.

**Comment by samuelgarcia** [comment]
Merci beaucoup Zach.
This is ok for me.

**Comment by zm711** [comment]
Hahah. Sam I think your merge of  #2305 without pre-commit freaked out the pre-commit setup. I just merged `main` into the PRs for two of my PRs and they both failed the pre-commit check. Was this your plan all along....

**Comment by samuelgarcia** [comment]
Oups. I am sorry. My eyes has develop some auto masking all around the precommit icon. I don't see it anymore.

**Comment by alejoe91** [comment]
@samuelgarcia please just leave it on. It's not a big deal really!

**Comment by alejoe91** [comment]
should be fixed in main with #2360, updating your branch to check if it fixes it here!

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**